vm/src/obj/objiter.rs

```diff
@@ -131,6 +131,20 @@ fn iter_next(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
                     Err(stop_iteration)
                 }
             }
+
+            PyObjectPayload::Bytes { ref value } => {
```

Comment:
Instead of inspecting the payload, could we use get_item here in some way with the current position?